analyzer/src/main/kotlin/MavenSupport.kt

```diff
@@ -329,7 +329,25 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
 
     fun parseVcsInfo(mavenProject: MavenProject): VcsInfo {
         return mavenProject.scm?.let { scm ->
-            val connection = scm.connection ?: ""
+            // If a Maven project does not define an SCM URL Maven takes the SCM URL of the parent project and appends
+            // the artifact ID of the child to this URL. This is fundamentally broken because it e.g. breaks all Git
+            // URLs and makes the bold assumption that the child is in a direct subdirectory of the parent. To fix this
+            // remove any paths appended by Maven by recursively taking the SCM URL of the parent project if it starts
+            // with the same string.
+            var parent = mavenProject.parent
+            var connection = scm.connection ?: ""
+            while(parent != null) {
+                val parentConnection = parent.scm?.connection
+                if (
+                    parentConnection != null && parentConnection.isNotBlank() && connection.startsWith(parentConnection)
+                ) {
+                    connection = parentConnection
+                    parent = parent.parent
+                } else {
+                    break
+                }
+            }
```

Comment:
If `mavenProject.scm` is `null` at the very beginning of this function we're not even entering this block due to `?.let`, and we're not searching for possible `<scm>` tags in any parent. So how about this instead:

    fun parseVcsInfo(mavenProject: MavenProject): VcsInfo {
        var scm = mavenProject.scm
        var parent = mavenProject.parent

        while (parent != null) {
            parent.scm?.let {
                it.connection?.let {
                    if (it.isNotBlank() && scm.connection.startsWith(it)) {
                        scm = parent.scm
                    }
                }
            }

            parent = parent.parent
        }

        if (scm == null) return VcsInfo.EMPTY

        val connection = scm.connection ?: ""
        val tag = scm.tag?.takeIf { it != "HEAD" } ?: ""

        val (type, url) = SCM_REGEX.matcher(connection).let {
            if (it.matches()) {
                val type = it.group("type")
                val url = it.group("url")

                // CVS URLs usually start with ":pserver:" or ":ext:", but as ":" is also the delimiter used by the
                // Maven SCM plugin, no double ":" is used in the connection string and we need to fix it up here.
                if (type == "cvs" && !url.startsWith(":")) {
                    Pair(type, ":" + url)
                } else {
                    Pair(type, url)
                }
            } else {
                Pair("", "")
            }
        }

        return VcsInfo(type, url, tag, "")
    }

Reply:
Done.